rocksdb/tests/test_options.py

```diff
@@ -67,20 +66,51 @@ def test_compression_opts(self):
         # default value
         self.assertEqual(isinstance(compression_opts, dict), True)
         self.assertEqual(compression_opts['window_bits'], -14)
-        # This doesn't match rocksdb latest
-        # self.assertEqual(compression_opts['level'], -1)
+        self.assertEqual(compression_opts['level'], 2**15 - 1)
         self.assertEqual(compression_opts['strategy'], 0)
         self.assertEqual(compression_opts['max_dict_bytes'], 0)
+        self.assertEqual(compression_opts['zstd_max_train_bytes'], 0)
+        self.assertEqual(compression_opts['parallel_threads'], 1)
+        self.assertEqual(compression_opts['enabled'], False)
 
         with self.assertRaises(TypeError):
-            opts.compression_opts = list(1,2)
+            opts.compression_opts = list(1, 2)
 
-        opts.compression_opts = {'window_bits': 1, 'level': 2, 'strategy': 3, 'max_dict_bytes': 4}
+        opts.compression_opts = {'window_bits': 1, 'level': 2, 'strategy': 3, 'max_dict_bytes': 4, 'zstd_max_train_bytes': 15, 'parallel_threads': 4, 'enabled': True}
         compression_opts = opts.compression_opts
         self.assertEqual(compression_opts['window_bits'], 1)
         self.assertEqual(compression_opts['level'], 2)
         self.assertEqual(compression_opts['strategy'], 3)
         self.assertEqual(compression_opts['max_dict_bytes'], 4)
+        self.assertEqual(compression_opts['zstd_max_train_bytes'], 15)
+        self.assertEqual(compression_opts['parallel_threads'], 4)
+        self.assertEqual(compression_opts['enabled'], True)
+
+    def test_bottommost_compression_opts(self):
+        opts = rocksdb.Options()
+        bottommost_compression_opts = opts.bottommost_compression_opts
+        # default value
+        self.assertEqual(isinstance(bottommost_compression_opts, dict), True)
+        self.assertEqual(bottommost_compression_opts['window_bits'], -14)
+        self.assertEqual(bottommost_compression_opts['level'], 2**15 - 1)
+        self.assertEqual(bottommost_compression_opts['strategy'], 0)
+        self.assertEqual(bottommost_compression_opts['max_dict_bytes'], 0)
+        self.assertEqual(bottommost_compression_opts['zstd_max_train_bytes'], 0)
+        self.assertEqual(bottommost_compression_opts['parallel_threads'], 1)
+        self.assertEqual(bottommost_compression_opts['enabled'], False)
+
+        with self.assertRaises(TypeError):
+            opts.compression_opts = list(1, 2)
+
+        opts.bottommost_compression_opts = {'window_bits': 1, 'level': 2, 'strategy': 3, 'max_dict_bytes': 4, 'zstd_max_train_bytes': 15, 'parallel_threads': 4, 'enabled': True}
+        bottommost_compression_opts = opts.bottommost_compression_opts
+        self.assertEqual(bottommost_compression_opts['window_bits'], 1)
```

Comment:
This is a bit repetitive and could be reduced (also, it is a good idea to double-check the dict passed is not kept):
```suggestion
        new_opts = {
            'window_bits': 1,
            'level': 2,
            'strategy': 3,
            'max_dict_bytes': 4,
            'zstd_max_train_bytes': 15,
            'parallel_threads': 4,
            'enabled': True,
        }
        opts.bottommost_compression_opts = new_opts
        self.assertIsNot(new_opts, opts.bottommost_compression_opts)
        for key, value in new_opts.values():
          self.assertEqual(opts.bottommost_compression_opts[key], value)
```

Reply:
Used this block.